src/optimization/dce.ml

```diff
@@ -131,6 +131,9 @@ let rec keep_field dce cf c is_static =
 		| Var { v_read = AccCall } -> check_accessor "get_"
 		| Var { v_write = AccCall } -> check_accessor "set_"
 		| _ -> false
+	|| match cf.cf_expr with
+		| Some ({ eexpr = TCall ({ eexpr = TField(ef, fa) }, args) }) -> not (PurityState.is_pure_field_access fa)
```

Comment:
What if we have a more complicated expression? I think if we want to be consistent here, we should utilize `has_side_effect`.

Reply:
I think that makes more sense. However, there are a ton of things being kept from the std library now with that approach. has_side_effect may be too pessimistic; I'll see if I can find a reasonable improvement there.

Reply:
Now that you mention it I think I tried this before and also concluded that it ended up keeping way too much stuff. Maybe this narrow pattern is a decent compromise in that regard.

Reply:
Maybe it would make sense to just not consider fields in the standard library as possible DCE entry points. If they aren't visited by some call that references them, they and their side effects probably aren't needed. I don't think we have any instances of this side effect issue in the standard library.